xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/source/MacroContentSourceReference.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.macro.source;
+
+import java.util.Objects;
+
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.xwiki.stability.Unstable;
+
+/**
+ * The reference of a wiki content to use as macro content.
+ * 
+ * @version $Id$
+ * @since 15.1RC1
+ * @since 14.10.5
+ */
+@Unstable
+public class MacroContentSourceReference
```

Comment:
Is it intentional that this is not shared with `CodeMacroSourceReference`? Seems like they have a lot in common.

Reply:
They are identical, actually. CodeMacroSourceReference have been removed, see the [platform side of this](https://github.com/xwiki/xwiki-platform/pull/2064/files#diff-09768dddce101f4e977b2940fa515838b75d308708e4b8f22044965306a69bb5).

Reply:
Indeed, missed that as it's on the other PR :)